File: python/app/models/item.py
"""Example Item model for CRUD operations."""

from datetime import datetime

from sqlalchemy import DateTime, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Item(Base):
    """
    Example Item entity - replace or extend for your domain.
    Demonstrates common patterns: id, name, description, timestamps.
    """

    __tablename__ = "items"

    id: Mapped[int] = mapped_column(primary_key=True, index=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False, index=True)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )

    def __repr__(self) -> str:
        return f"<Item(id={self.id}, name={self.name})>"

File: go/internal/models/item.go
package models

import "time"

// Item represents an item entity - replace or extend for your domain.
type Item struct {
	ID          int        `json:"id"`
	Name        string     `json:"name"`
	Description *string    `json:"description,omitempty"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
}

// ItemCreate is the payload for creating an item.
type ItemCreate struct {
	Name        string  `json:"name"`
	Description *string `json:"description,omitempty"`
}

// ItemUpdate is the payload for partial updates.
type ItemUpdate struct {
	Name        *string `json:"name,omitempty"`
	Description *string `json:"description,omitempty"`
}
